content/posts/rust-pyo3-support-in-pyodide.md

```diff
@@ -0,0 +1,338 @@
+---
+title: "Rust/PyO3 Support in Pyodide"
+date: 2022-04-15T12:37:52-07:00
+author: "Hood Chatham"
+# author: ["Me", "You"] # multiple authors
+showToc: true
+TocOpen: false
+draft: false
+hidemeta: false
+comments: false
+description: "Getting the pyca/cryptography package to run in Pyodide"
+# canonicalURL: "https://canonical.url/to/page"
+disableHLJS: true # to disable highlightjs
+disableShare: false
+hideSummary: false
+searchHidden: true
+ShowReadingTime: true
+ShowBreadCrumbs: true
+ShowPostNavLinks: true
+cover:
+    # image: "<image path/url>" # image path/url
+    # alt: "<alt text>" # alt text
+    # caption: "<text>" # display caption under cover
+    relative: false # when using page bundles set this to true
+    hidden: true # only hide on current single page
+---
+
+
+We are happy to announce that the next version of Pyodide will ship with the
+newest version of the Cryptography package, including its Rust extension module.
+The Cryptography package was one of the first major packages that included a
+Rust extension module. Python binary extensions in Rust for new projects have
+been gaining popularity lately. The Cryptography package has been one of the
+most often requested packages in the Pyodide issue tracker so it is a priority
+for us to support it.
+
+See [this talk by the Cryptography
+maintainers](https://www.youtube.com/watch?v=z_Eiy2W0APU) for an interesting
+discussion of their reasons for using Rust and the problems in the ecosystem
+that needed to be fixed before they could use it.
+
+We want to build the Rust extension module for our WebAssembly-based
+distribution. Rust has good support for the <code
+class="pkg">wasm32-unknown-unknown</code> target and has popular tools like
+<code class="pkg">wasm-bindgen</code>. However, <code
+class="pkg">wasm32-unknown-unknown</code> uses a custom "wasm ABI" [which is not
+compatible with C/C++
+code](https://github.com/rustwasm/team/issues/291#issuecomment-644946504l). To
+use Rust in a project that also includes C/C++ code we need to use the <code
+class="pkg">wasm32-unknown-emscripten</code> target. Unfortunately, the <code
+class="pkg">wasm32-unknown-emscripten</code> target does not work all that well,
+due to limited development resources and difficulties coordinating between the
+Emscripten and Rust projects.
+
+
+In this blog post I will give a technical description of many of the problems I
+ran into in the process of building Rust extension modules for Pyodide.
```

Comment:
Maybe it's nice to be clear that you did manage to solve them all and readers will learn about the solutions too?

```suggestion
In this blog post I will give a technical description of many of the challenges I
ran into in the process of building Rust extension modules for Pyodide and their solutions.
```